core/src/main.py

```diff
@@ -54,15 +17,74 @@
 # switch into unprivileged mode, as we don't need the extra permissions anymore
 utils.set_mode_unprivileged()
 
-# run main event loop and specify which screen is the default
-from apps.homescreen.homescreen import homescreen
-from apps.management.recovery_device.homescreen import recovery_homescreen
-from apps.common.storage import device
 
-if device.is_recovery_in_progress():
-    # TODO: @jpochyla
+def _boot_recovery():
+    # load applications
+    import apps.management
+
+    if __debug__:
+        import apps.debug
+
+    # boot applications
+    apps.management.boot()
+
+    if __debug__:
+        apps.debug.boot()
```

Comment:
Why isn't this under a single `if __debug__` statement?

Reply:
I don't know, it can be. It's just that first we import the apps and then we boot them 🤷‍♂